runtime/capsules/src/mctp/mux.rs

```diff
@@ -338,14 +265,100 @@ impl<'a, M: MCTPTransportBinding<'a>> MuxMCTPDriver<'a, M> {
             })
     }
 
-    fn get_total_hdr_size(&self) -> usize {
-        MCTP_HDR_SIZE + self.mctp_device.get_hdr_size()
+    fn send_next_packet(&self, cur_sender: &'a MCTPTxState<'a, M>) {
+        let mut tx_pkt = SubSliceMut::new(self.tx_pkt_buffer.take().unwrap());
+        let mctp_hdr_offset = self.mctp_hdr_offset();
+        let pkt_end_offset = self.get_mtu();
+
+        // set the window of the subslice for MCTP header and the payload
+        tx_pkt.slice(mctp_hdr_offset..pkt_end_offset);
+
+        match cur_sender.fill_next_packet(&mut tx_pkt, self.local_eid.get()) {
+            Ok(len) => {
+                tx_pkt.reset();
+                match self
+                    .mctp_device
+                    .transmit(tx_pkt.take(), len + mctp_hdr_offset)
+                {
+                    Ok(_) => (),
+                    Err((err, buf)) => {
+                        println!("MuxMCTPDriver: Failed to transmit {:?}", err);
+                        self.tx_pkt_buffer.replace(buf);
+                    }
+                }
+            }
+            Err(err) => {
+                println!("MuxMCTPDriver: Failed to start transmit {:?}", err);
+                self.tx_pkt_buffer.replace(tx_pkt.take());
+            }
+        }
+    }
+
+    fn process_first_packet(
+        &self,
+        mctp_hdr: MCTPHeader<[u8; MCTP_HDR_SIZE]>,
+        msg_type: MessageType,
+        pkt_payload: &[u8],
+    ) {
+        let rx_state = self
+            .receiver_list
+            .iter()
+            .find(|rx_state| rx_state.is_receive_expected(msg_type));
+
+        if let Some(rx_state) = rx_state {
+            rx_state.start_receive(mctp_hdr, msg_type, pkt_payload);
+        } else {
+            println!("MuxMCTPDriver: No matching receive request found. Dropping packet.");
+        }
+    }
+
+    fn process_packet(&self, mctp_hdr: MCTPHeader<[u8; MCTP_HDR_SIZE]>, pkt_payload: &[u8]) {
+        if self.local_eid != mctp_hdr.dest_eid().into() {
+            println!("MuxMCTPDriver: Packet not for this Endpoint. Dropping packet.");
+            return;
+        }
+
+        if mctp_hdr.eom() != 1 && pkt_payload.len() < 64 {
+            println!("MuxMCTPDriver: Received first or middle packet with less than 64 bytes. Dropping packet.");
```

Comment:
Is it required that the packets be fuller than that?

Reply:
I was trying to ensure the following clause. But I think this check should be on the first packet of the multi packet msg. 
The middle packets are already checked against the first packet length in `is_next_packet`. I've adjusted the code accordingly.

Ref: [MCTP baseline specification](https://www.dmtf.org/sites/default/files/standards/documents/DSP0236_1.3.3.pdf)
   Section  8.4.2  Baseline transmission unit 
`The transmission unit of all packets in a given message shall be the same size, except for the transmission unit in the last packet (packet with EOM bit = 1b). Except for the last packet, this size shall be at least the baseline transmission unit size.`